### ISSUE TITLE
after adding .json file to board manager URLSs the IDE gives index errror about avrdude not found

### ISSUE THREAD

**Issue by gitterman** [SpenceKonde/ATTinyCore#646]
After I have added http://drazzy.com/package_drazzy.com_index.json to my board manager URLs I get the following error when I start the IDE
Index error: could not find referenced tool name=avrdude version=6.3.0-arduino17or18 packager=arduino

Looking for avrdude.exe I find it in %LOCALAPPDATA%\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17\bin as well as in %LOCALAPPDATA%\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino18\bin, though.

**Comment by prandeamus** [comment]
I saw similar messages yesterday loading the Arduino IDE under linux, although I didn't have time to investigate.

EDIT I've had that JSON configured for some months and it wasn't a change on my part, as far as I am aware.

**Comment by SpenceKonde** [comment]
Edit oh crap it's giving this message even for cores that don't use that version of AVRdude?! uuuuuuggghhhhh

@per1234 what do I dooooooo? I can't fix the problem or I will give some users null pointer exceptions, but I can't leave it like it is because everyone is complaining about these messages....

**Comment by per1234** [comment]
>it's giving this message even for cores that don't use that version of AVRdude?

It occurs under the following conditions:

- Using Arduino IDE 1.8.19 or earlier
  - The only time I have been able to produce the error with Arduino CLI or Arduino IDE 2.x is when attempting to install `megaTinyCore:megaavr@2.5.8` (which fails immediately without any harmful effects and is likely to be a rare occurrence anyway)
- http://drazzy.com/package_drazzy.com_index.json is in the "Additional Boards Manager URLs" field
- Library Manager or Boards Manager is opened or closed (any time the indexes are loaded)

So it occurs regardless of whether megaTinyCore, or any other core is installed. It's purely associated with the presence of the URL in the user's preferences.

>what do I dooooooo?

I'm going to take another look at it ASAP. I apologize for kind of leaving things hanging earlier. I had to switch my focus over to work duties. At the least, I'll write a summary of the NPE situation on https://github.com/SpenceKonde/ReleaseScripts/pull/37
